### PR TITLE
Feature/aos2 22 init config

### PR DESCRIPTION
**Pull request by krivanek06**
- [x] - Link up linting
- [x] - Add linting to husky
- [x] - Set default schematic behaviors
- [x] - Component: OnPush, display block

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -1,4 +1,4 @@
 #!/usr/bin/env sh
 . "$(dirname -- "$0")/_/husky.sh"
 
-npx lint-staged --relative 
+npx pretty-quick --staged
```
Comment (m-thompson-code) [inline .husky/pre-commit:4]: Does this override the existing linted-staged? Ideally we would keep our existing hook and just add prettier

File: .prettierignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
+package.json
+package-lock.json
+yarn.lock
+dist
```
Comment (m-thompson-code) [inline .prettierignore:4]: Could we add `coverage` and maybe look over .gitignore for other directories to skip?

File: package.json [hunk 3]
```diff
@@ -11,6 +11,17 @@
     "lint": "ng lint --fix",
     "deploy": "ng deploy --repo=https://github.com/bitovi/opensource-analytics-dashboard.git --base-href=https://bitovi.github.io/opensource-analytics-dashboard/"
   },
+  "lint-staged": {
+    "src/**/*.{js,ts,scss,md,html,json}": [
+      "prettier --write",
+      "git add"
+    ]
+  },
+  "husky": {
```
Comment (m-thompson-code) [inline package.json:20]: I don't believe these do anything since we're on husky v8

**Review by m-thompson-code — CHANGES_REQUESTED** [review]
I believe that the package.json is using an older setup for husky. Also, the changed to the pre-comment hook is removing the existing hook (which is how we're running our formatter)

Also because of the nature of this commit (linting affects basically every file), I just wanted to confirm that this PR is only really changing `.husky`, `package-lock.json`, `package.json` and a couple added files.

**Comment by m-thompson-code** [comment]
I believe you should be able to put the command in `.lintstagedrc.json`:

```json
{
    "*.{js,ts,html}": ["npx ng lint opensource-analytics-dashboard --fix=true --lint-file-patterns"],
     "*.{js,ts,html, scss}": ["pretty-quick --staged]
}
```

I don't know for sure how many file extensions should use prettier lol, but yea something like this maybe?

File: .husky/pre-commit (hunk 1 above)
Comment (krivanek06) [reply .husky/pre-commit:4]: Do you mean that by keeping both hooks in pre-commit file? 
so the content will be
`npx lint-staged --relative
npx pretty-quick --staged`

File: .prettierignore (hunk 2 above)
Comment (krivanek06) [reply .prettierignore:4]: I am considering adding the following:
`
coverage
node_modules
.vscode/*
/.angular/cache
`

**Comment by krivanek06** [comment]
I left only `npx lint-staged --relative` in the pre-commit hook and moved the `pretty-quick --staged` into `.lintstagedrc.json`. Remove `husky hooks` in package.json since we have .husky directory

**Comment by krivanek06** [comment]
So the added files were:
- .prettierrc.json
- .prettierignore
- .eslintrc.json - plugin "plugin:prettier/recommended"
- packages: `husky`, `prettier`

I believe that should be it.
Everything else is just formatting files

**Comment by m-thompson-code** [comment]
https://gist.github.com/LayZeeDK/c822cc812f75bb07b7c55d07ba2719b3#file-angular-cli-node-js-typescript-rxjs-compatiblity-matrix-csv-L43

Project doesn't include engines property to safeguard node/npm versions. We should go with 14/ npm 6 or node 16 / npm 7, I'm going to suggest node 16 for now

**Review by m-thompson-code — APPROVED** [review]
oops, restarted review haha. I just meant to approve. +1 looks good to me. merging

**Pull request merged**